capstone/src/test/javascript/mapsSpec.js

```diff
@@ -0,0 +1,22 @@
+
+describe('Map Marker and InfoWindow Display', () => {
+  it('will add map content using data from JSON',
+      async () => {
+        const mapsJsonData = [
+          {'name': 'New York University',
+            'city': 'New York',
+            'isMainCampus': true,
+            'latitude': 40.729452,
+            'longitude': -73.997264},
+          {'name': 'Stanford University',
+            'city': 'Stanford',
+            'isMainCampus': true,
+            'latitude': 37.429434,
+            'longitude': -122.167359},
+        ];
+        spyOn(window, 'fetch')
+            .and.returnValue(Promise.resolve({json: () => mapsJsonData}));
+        await fetchData();
+        // TODO(biancamacias): mock? assert functions work? but how?
```

Comment:
So far I think the set up is pretty good. One testable aspect here is making sure that the constructors/methods are called properly (which would imply the Google Maps datatypes are being created/manipulated properly). Another would be any helper functions, like `setDescription`.

For the first aspect, you would spy on the constructor and the methods like this:
```javascript
spyOn(theClass.prototype, 'constructor');
spyOn(theClass.prototype, 'method');
// do something with theClass
expect(theClass.prototype.constructor)
  .toHaveBeenCalledWith(params);
expect(theClass.prototype.method)
  .toHaveBeenCalledWith(otherParams);
```

The second aspect should be more directly testable, as you won't need external functions, and can directly compare the actual result to the expected result:
```javascript
expect(helper()).toEqual(expectedResult());
```

If you want, we can discuss this further offline.